tools/wpt/browser.py

```diff
@@ -516,264 +516,201 @@ def version(self, binary=None, webdriver_binary=None):
-        self.logger.info("Downloading Chrome from %s" % url)
-        resp = get(url)
-        installer_path = os.path.join(dest, filename)
-        with open(installer_path, "wb") as f:
-            f.write(resp.content)
-        return installer_path
-
-    def install(self, dest=None, channel=None):
-        if channel != "nightly":
-            raise NotImplementedError("We can only install Chrome Nightly (Chromium ToT) for you.")
-        dest = self._get_dest(dest, channel)
-
-        installer_path = self.download(dest, channel)
-        with open(installer_path, "rb") as f:
-            unzip(f, dest)
-        os.remove(installer_path)
-        return self.find_nightly_binary(dest)
-
-    def install_mojojs(self, dest, channel, browser_binary):
-        if channel == "nightly" or channel == "canary":
-            url = self._latest_chromium_snapshot_url() + "mojojs.zip"
-        else:
-            chrome_version = self.version(binary=browser_binary)
-            assert chrome_version, "Cannot determine the version of Chrome"
-            # Remove channel suffixes (e.g. " dev").
-            chrome_version = chrome_version.split(' ')[0]
-            url = "https://storage.googleapis.com/chrome-wpt-mojom/%s/linux64/mojojs.zip" % chrome_version
+    }.get(uname[0])
 
-        extracted = os.path.join(dest, "mojojs", "gen")
-        last_url_file = os.path.join(extracted, "DOWNLOADED_FROM")
-        if os.path.exists(last_url_file):
-            with open(last_url_file, "rt") as f:
-                last_url = f.read().strip()
-            if last_url == url:
-                self.logger.info("Mojo bindings already up to date")
-                return extracted
-            rmtree(extracted)
+    # The last revision number used to successfully download/install browser binary or webdriver.
+    last_revision_used = None
 
-        self.logger.info("Downloading Mojo bindings from %s" % url)
-        unzip(get(url).raw, dest)
-        with open(last_url_file, "wt") as f:
-            f.write(url)
-        return extracted
+    # The last revision number used to try and download/install browser binary or webdriver.
+    # Could have been successful or unsuccessful.
+    revision_attempted = None
 
     def _chromedriver_platform_string(self):
-        platform = self.platforms.get(uname[0])
-
-        if platform is None:
+        if self.platform is None:
             raise ValueError("Unable to construct a valid Chrome package name for current platform")
-        platform = platform.lower()
 
-        if platform == "linux":
+        if self.platform == "Linux":
             bits = "64" if uname[4] == "x86_64" else "32"
-        elif platform == "mac":
+        elif self.platform == "Mac":
             bits = "64"
-        elif platform == "win":
+        elif self.platform == "Win":
             bits = "32"
 
-        return "%s%s" % (platform, bits)
+        return f"{self.platform.lower()}{bits}"
 
     def _chromium_platform_string(self):
```

Comment:
Will this be used by Chrome? If it's just for Chromium, can it be moved into that class?

Reply:
Since it is not possible to install a Chrome browser binary via `./wpt install`, then this will be unlikely to be used by Chrome. However, if that were ever to change and we added functionality to install Chrome with wpt, then this would possibly be shared by Chrome and Chromium. Is it just best to isolate all install functionality and attribute it only to Chromium, since as of now we technically _shouldn't_ be installing a chromedriver for Chrome?
I do think that's a good idea, and then if something changes and we supply Chrome installs, there can be some minor refactoring for what is shared. Since supplying Chrome install not on any horizon at the moment, I like this approach.